site/docs/components/password_input.mdx

```diff
@@ -0,0 +1,256 @@
+</LargeParagraph>
+
+## Principles
+- A label should always be provided with password fields.
+- If possible, allow the password input to reflect the strength of the password entered.
+- If possible, add other programmatic assistance. Detect and suggest changes to provide the user with assistance in generating strong passwords.
+- If possible, do input validation in client-side real-time and provide the user with immediate feedback. See [HDS Form validation pattern page](/patterns/form-validation) for more information and best practices with password fields.
+- Password fields should always be mandatory and it is recommended to mark it as a required field.
+- Use `autocomplete="new-password"` and `id="new-password"` for the password input in a sign-up form, and for the new password in a reset-password form.
+- HDS password input provides a way to mask/unmask the password. Do not disabled this functionality without a good reason.
+- Use `aria-label` and `aria-describedby` for password inputs.
+
+## Accessibility
+- In the case of an insufficient password, aim to provide clear instructions to the user on how to correct the mistake. Always provide a text description of the error.
+- It is required to use `autocomplete` attribute if input control is listed in the common input purposes list. You can read the full list of common input purposes at <Link href="https://www.w3.org/TR/WCAG21/#input-purposes" external>WCAG 2.1 documentation website</Link>. You can read more about the requirement from <Link href="https://www.w3.org/WAI/WCAG21/Understanding/identify-input-purpose" external>WCAG 2.1 Identify Input Purpose guideline</Link>.
+- Place sign-in in its own `form` element and use `landmark` region to mark the login area. This allows assistive technologies to quickly locate and skip to the login section.
+
+## Security
+As HDS password input provides only the front-end solution, remember that you need to take care of the security in your project.
+Here are some good practices you need to consider.
+
+### General Security
+- Do not prevent the user from copying and pasting the password into the field
+- Prefer using ready authentication libraries and services, and familiarize yourself properly with their implementation
+- In City of Helsinki Services, always use Helsinki Profiili authentication if possible.
+- Never store the password in a readable form. Always use proper hash algorithms.
+- Always protect the communication with secure connections (TLS 1.2 or newer).
+- Encourage usage of multi-factor authentication and password managers.
+
+### Creating a new password
+- Inform user properly what are the requirements for their password.
+- Do not accept password matching user identifier, e.g. username or email.
+- Do not truncate, sanitize, or format in any way password user has given.
+- Set minimum length for password 12 to 16 characters.
```

Comment:
Is there any way we could make "12" the default min. length (while possibly allowing overriding it when really needed?)

Reply:
This could surprise the component users that there is a limit for it. Also validating this requires extra work, so this is also a possible future enhancement.